### PR TITLE
Add aten::norm operator and layer test

### PR DESCRIPTION
**Pull request by sikorsl1**
Details:

- `aten::norm` operator has been added
- layer test for this operator has been added

Ticket:

- 94285

### CODE REVIEW

File: src/frontends/pytorch/src/op/norm.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+OutputVector translate_norm(NodeContext& context) {
+    auto input_tensor = context.get_input(0);
+    auto p = context.const_input<int64_t>(1);
+    auto dim = context.get_input(2);
+    auto keep_dim = context.const_input<bool>(3);
+
+    std::shared_ptr<ov::op::util::ArithmeticReductionKeepDims> reduce;
+
+    FRONT_END_OP_CONVERSION_CHECK(p == 1 || p == 2, "OpenVino export only p-norms with p of 1 or 2");
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/norm.cpp]:
```suggestion
    FRONT_END_OP_CONVERSION_CHECK(p == 1 || p == 2, "OpenVino export supports only p-norms with p of 1 or 2");
```

File: tests/layer_tests/pytorch_tests/test_norm.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from pytorch_layer_test_class import PytorchLayerTest
+import numpy as np
+import torch
+
+
+@pytest.mark.parametrize('p', [1, 2])
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_norm.py]: I think we can support other norms too using formulas from https://pytorch.org/docs/stable/generated/torch.linalg.norm.html#torch.linalg.norm. Maybe not at best performance though.

Comment (sikorsl1) [reply tests/layer_tests/pytorch_tests/test_norm.py]: I also thought about it, but I decided to use the following approach:
https://github.com/pytorch/pytorch/blob/master/torch/onnx/symbolic_opset9.py#L3316:~:text=feature_dropout-,%40,),-%40
But I can broaden my implementation to the case, where `p` is different than 1 or 2.

File: src/frontends/pytorch/src/op/norm.cpp (hunk 1 above)
Comment (sikorsl1) [reply src/frontends/pytorch/src/op/norm.cpp]: done

File: tests/layer_tests/pytorch_tests/test_norm.py (hunk 2 above)
Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_norm.py]: Yes, please support other norms, we do not need to restrict ourselves with onnx conversion

Comment (sikorsl1) [reply tests/layer_tests/pytorch_tests/test_norm.py]: All types of numeric norms are now supported. Frobenius and nuclear norms are separate operators - namely `aten::nuclear_norm` and `aten::frobenius_norm`.

**Review by mvafin — APPROVED** [review]

**Pull request merged**